### PR TITLE
Get instance action fix

### PR DESCRIPTION
**Pull request by snigle**
For gophercloud#1845

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://docs.openstack.org/api-ref/compute/?expanded=show-server-action-details-detail,list-actions-for-server-detail#show-server-action-details

### CODE REVIEW

File: openstack/compute/v2/extensions/instanceactions/results.go [hunk 1]
```diff
@@ -58,13 +58,77 @@ func ExtractInstanceActions(r pagination.Page) ([]InstanceAction, error) {
...
+
+	e.StartTime = time.Time(s.StartTime)
+	e.FinishTime = time.Time(s.FinishTime)
+
+	return err
+}
+
+// InstanceActionDetail gives more details on instance action.
+type InstanceActionDetail struct {
+	Action       string    `json:"action"`
```
Comment (snigle) [inline openstack/compute/v2/extensions/instanceactions/results.go]: tried with nested type InstanceAction but the func extractIntoPtr didn't work, So I copy paste all field :/

**Comment by jtopjian** [comment]
@snigle What's the purpose of this PR? The title has "fix" in it, but no other details.

Also, instead of API documentation, can you provide the server-side service code links which prove this implementation is correct?

**Comment by snigle** [comment]
Sorry I added two missing field in the Get instance actions. (event and
updated At.)
I didn't see these field when I wrote the first version.

**Comment by jtopjian** [comment]
@snigle Thanks. It looks like this set of API calls has a number of features added/modified via microversions. The way that some of these features are implemented in this PR aren't following our current [guidelines](https://github.com/gophercloud/gophercloud/blob/master/docs/MICROVERSIONS.md) for microversions. For example, there should be dedicated extract methods for `Host` and `HostID`. 

However, I'm currently refactoring things to simplify our guidelines (https://github.com/gophercloud/gophercloud/pull/1854). Once this is merged, you'll have to make some minor changes, which I'll outline in a review.

**Comment by snigle** [comment]
> @snigle Thanks. It looks like this set of API calls has a number of features added/modified via microversions. The way that some of these features are implemented in this PR aren't following our current [guidelines](https://github.com/gophercloud/gophercloud/blob/master/docs/MICROVERSIONS.md) for microversions. For example, there should be dedicated extract methods for `Host` and `HostID`.
> 
> However, I'm currently refactoring things to simplify our guidelines (#1854). Once this is merged, you'll have to make some minor changes, which I'll outline in a review.

Ok I updated the code as the same way of your PR. 
https://github.com/gophercloud/gophercloud/pull/1851/commits/b5400309a6b3e2fc95d8b27ac9ba541129f97192

**Comment by jtopjian** [comment]
@snigle Nice - thank you! Everything looks correct. The only remaining item is fixing the Travis errors:

https://travis-ci.org/gophercloud/gophercloud/jobs/651404710#L492-L495

Because some fields were changed to pointers, some of the fixtures need modified, too. For example:

https://github.com/gophercloud/gophercloud/pull/1851/files#diff-a5d110494cfe4ed97e9f81b89d0e752eR76

```go
var updatedAt = time.Date(2018, 04, 25, 1, 26, 36, 0, time.UTC)
var GetExpected = instanceactions.InstanceActionDetail{
  ...
  UpdatedAt: &updatedAt,
}
```

**Comment by snigle** [comment]
> @snigle Nice - thank you! Everything looks correct. The only remaining item is fixing the Travis errors:
> 
> https://travis-ci.org/gophercloud/gophercloud/jobs/651404710#L492-L495
> 
> Because some fields were changed to pointers, some of the fixtures need modified, too. For example:
> 
> https://github.com/gophercloud/gophercloud/pull/1851/files#diff-a5d110494cfe4ed97e9f81b89d0e752eR76
> 
> ```go
> var updatedAt = time.Date(2018, 04, 25, 1, 26, 36, 0, time.UTC)
> var GetExpected = instanceactions.InstanceActionDetail{
>   ...
>   UpdatedAt: &updatedAt,
> }
> ```

Oups .... Forgotten to run test locally :s Fixed now :)

**Pull request merged**